tasks/selinux_load_module.yml

```
@@ -1,30 +1,41 @@
 ---
-- name: Install module
+- name: Prepare module installation
   when: state == "enabled" and item.path is defined
   block:
-    - name: Create temporary directory
-      tempfile:
-        state: directory
-        prefix: linux_system_role.selinux
-      register: tempdir
-
-    - name: Copy module file {{ item.path }}
-      copy:
-        dest: "{{ tempdir.path }}"
-        src: "{{ item.path }}"
-        mode: preserve
-
-    - name: Install the module
-      local_semodule:
-        path: "{{ tempdir.path }}/{{ item.path | basename }}"
-        priority: "{{ priority }}"
-        state: enabled
-      notify: __selinux_reload_policy
-  always:
-    - name: Clean temporary directory {{ tempdir.path }}
-      file:
-        path: "{{ tempdir.path }}"
-        state: absent
+    - name: Get checksum for {{ item.path }}
+      stat:
+        path: "{{ item.path }}"
+        checksum_algorithm: sha256
+      register: module_file
```

```suggestion
      register: module_file
      delegate_to: localhost
```
because `item.path` is on the controller, not the managed node

---

Thanks. I tested it only using `-c local -i localhost,` and therefore it passed even though it was not correct.

---

ok - let me know if you want me to show you how to do local qemu/kvm testing using tox-lsr like https://github.com/linux-system-roles/tox-lsr#qemu-testing

---

I'll take a look. I generally have no problem to test the code on different nodes. But I'm not sure how to run all the CI linters locally.